### PR TITLE
Add Google Analytics

### LINKED ISSUE #961: Enhance Data Being Sent to Google Analytics
### User Story
As a LaterPay WordPress merchant, I would like to be able to view all of my data integrated together in one place so that I am confident in how my LaterPay implementation is performing.

### Acceptance Criteria
WordPress needs to capture the same information that LaterPay is collecting from other integration options.
- WordPress plugin detects if GA is already implemented by the merchant
- If WordPress plugin detects existing GA, we add a tracker
- If we don't detect existing GA, we inject GA script tag to LaterPay-enabled pages (ie: pages that have a price associated to them)
- We emit events to GA for all configured trackers on the page, so that both the publisher and LaterPay get events
- Merchant can opt-out out of the LaterPay events being sent to GA
- See mockup below for suggested design & copy

### Technical Details
**Information on Data and Recipients:**
- UA-50448165-9 (LaterPay USA in Live Mode)
- UA-50448165-10 (LaterPay USA in Sandbox mode)
- UA-50448165-3 (LaterPay EU in Live Mode)
- UA-50448165-4 (LaterPay EU in Sandbox mode)
- UA-XXXXXX (whatever the publisher has configured; see mockup for input location)

**Event Actions**
- Paid Content Replacement Show (triggered when closed paywall is viewed)
- Paid Content Purchase (triggered when user clicks buy button and is redirected to our hosted purchase dialog pages)
- Paid Content Purchase Complete (triggered when customer returns after successful purchase and content is unlocked when access state is true)
- Paid Content Identify (triggered when user clicks "I have already purchased this" link)

**Event Category**
"LaterPay WordPress Plugin"

**Event Label** 
these events should have in comma-separated form:
- "the_title()" (the WordPress title of the post)
- "bloginfo()" (the name of the site)
- "get_permalink()" (the permalink of the post)

### Related Documentation
How this is accomplished in other LaterPay integration methods:
* https://docs.google.com/spreadsheets/d/1sCpMX4Mnru97xkeHncvNVKK4mO5vhNJWXuyN4u3qpWY/edit#gid=0
* http://docs.laterpay.net/connector/analytics/integration/
* http://docs.laterpay.net/connector/analytics/event_actions/

### UX
image

If a user un-checks the second "LaterPay Google Analytics" box, please display the following in a modal:

> LaterPay collects this information to improve our products and services and also so that you can determine the effectiveness of your pricing strategy using our Merchant Analytics dashboard.
> 
> Are you sure you would like to disable this feature?

With a button to "Yes, Disable Tracking" and a Cancel link (similar to Save & Cancel on the Pricing tab)

### PR DESCRIPTION
**Pull request by thrijith**
Fixes #961

### CODE REVIEW

File: laterpay/application/Controller/Admin/Settings.php [hunk 1]
```diff
@@ -43,6 +43,35 @@ public function load_assets()
...
+        wp_enqueue_script( 'laterpay-backend-options' );
+
+        // Localize string to be used in script.
+        wp_localize_script(
+            'laterpay-backend-options',
+            'lpVars',
+            array(
+                'modal' => array(
+                    'id'    => 'lp_ga_modal_id',
+                    'title' => 'Disable Tracking'
```
Comment (PatelUtkarsh) [inline laterpay/application/Controller/Admin/Settings.php]:
```suggestion
                    'title' => esc_html__( 'Disable Tracking' )
```

File: laterpay/application/Controller/Admin/Settings.php [hunk 2]
```diff
@@ -981,4 +1011,247 @@ public function get_laterpay_pro_merchant_description() {
...
     }
+
+    /**
+     * Method to render ga fields.
+     *
+     * @param array $fields array of field params.
+     *
+     * @return void
+     */
+    public function get_ga_field_markup( $fields = null ) {
```
Comment (PatelUtkarsh) [inline laterpay/application/Controller/Admin/Settings.php]: @thrijith Should this be part of view?

File: laterpay/application/Controller/Install.php [hunk 3]
```diff
@@ -519,4 +520,57 @@ public function update_default_pricing_behaviour() {
...
+     * Add options needed for sending Google Analytics Events.
+     *
+     * @since 2.2.0
+     *
+     * @return void
+     */
+    public function init_ga_options() {
+        $current_version = get_option( 'laterpay_plugin_version' );
+        // @todo: change version compared to `2.1.0`
+        // @todo: Only Add / Update option only if current one doesn't exist.
```
Comment (PatelUtkarsh) [inline laterpay/application/Controller/Install.php]: Address this todo? Confirm with the Ksmoore125 for version.

File: laterpay/application/Controller/Install.php [hunk 4]
```diff
@@ -519,4 +520,57 @@ public function update_default_pricing_behaviour() {
...
+        }
+
+        $lp_personal_data = [];
+        $lp_tracking_data = [];
+
+        $lp_personal_data['laterpay_ga_personal_enabled_status'] = 0;
+        $lp_personal_data['laterpay_ga_personal_ua_id']          = '';
+
+        if ( is_multisite() ) {
+            $monster_insights_network_profile = get_site_option( 'monsterinsights_network_profile', array() );
```
Comment (PatelUtkarsh) [inline laterpay/application/Controller/Install.php]: Should add plugin URL somewhere in a comment and what we are doing if that plugin is activated.

File: laterpay/application/Controller/Install.php [hunk 5]
```diff
@@ -519,4 +520,57 @@ public function update_default_pricing_behaviour() {
...
+
+        if ( ! empty( $monster_ua_id ) ) {
+            $lp_personal_data['auto_detected']              = 1;
+            $lp_personal_data['laterpay_ga_personal_ua_id'] = $monster_ua_id;
+        }
+
+        // Add Personal Google Analytics Data.
+        update_option( 'laterpay_user_tracking_data', $lp_personal_data );
+
+        $lp_is_plugin_live = (bool) get_option( 'laterpay_plugin_is_in_live_mode' );
```
Comment (PatelUtkarsh) [inline laterpay/application/Controller/Install.php]: If this is used multiple times then we can create a generic function for this and can then use it directly in if condition.

File: laterpay/asset_sources/js/laterpay-post-view.js [hunk 6]
```diff
@@ -207,7 +214,15 @@
...
                     });
             },
-
+            bindAlreadyPurchasedEvents = function() {
+              // handle clicks on already bought link.
+              $o.body
+              .on('click', $o.lp_already_bought, function(e) {
+                e.preventDefault();
+                sendGAEvent( 'Paid Content Identify' );
+                window.location.href = $(this).attr( 'href' );
```
Comment (PatelUtkarsh) [inline laterpay/asset_sources/js/laterpay-post-view.js]: @thrijith Did you test with adblocker on?

File: laterpay/application/Controller/Admin/Settings.php (hunk 2 above)
Comment (thrijith) [reply laterpay/application/Controller/Admin/Settings.php]: This is modelled after https://github.com/laterpay/laterpay-wordpress-plugin/blob/86011b7bcbfbde68f5c46a8a38e1c99ea0f5095d/laterpay/application/Controller/Admin/Settings.php#L581-L582

and is a callback for `add_settings_field `, GA fields needed customizations so created a new one for it. It generates markup based on field/values passed.

Let me know if this needs to be changed.

File: laterpay/asset_sources/js/laterpay-post-view.js (hunk 6 above)
Comment (thrijith) [reply laterpay/asset_sources/js/laterpay-post-view.js]: My ad-blocker was on. Will check and verify once to be thorough.

File: laterpay/application/Controller/Admin/Settings.php (hunk 2 above)
Comment (PatelUtkarsh) [reply laterpay/application/Controller/Admin/Settings.php]: Is there any model for backend UI? If then we should try to move if possible.

Comment (thrijith) [reply laterpay/application/Controller/Admin/Settings.php]: Nope, not right now. Everything on the setting page is added via https://github.com/laterpay/laterpay-wordpress-plugin/blob/86011b7bcbfbde68f5c46a8a38e1c99ea0f5095d/laterpay/application/Controller/Admin/Settings.php#L86

File: laterpay/asset_sources/js/laterpay-post-view.js (hunk 6 above)
Comment (thrijith) [reply laterpay/asset_sources/js/laterpay-post-view.js]: Tested with [this](https://chrome.google.com/webstore/detail/adblock-plus/cfhdojbkjhnklbpkdaibdccddilifddb) enabled. Let me know if needed to test with something else.

**Pull request merged**